### PR TITLE
move spark integration page to focus on assets

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: docs/content/integrations/spark.mdx [hunk 1]
```diff
@@ -1,62 +1,60 @@
...
+There are two approaches to writing Dagster assets and ops that invoke Spark computations:
 
-## Op body submits job
+## Asset body submits Spark job
 
-With this approach, the code inside the op submits a Spark job to an external system like Databricks or EMR, usually pointing to a jar or zip of Python files that contain the actual Spark data transformations and actions.
+With this approach, the code inside the asset definition submits a Spark job to an external system like Databricks or EMR, usually pointing to a jar or zip of Python files that contain the actual Spark data transformations and actions.
 
-If you want to use this approach to run a Spark job on Databricks, <PyObject module='dagster_databricks' object='create_databricks_run_now_op' /> or <PyObject module='dagster_databricks' object='create_databricks_submit_run_op' /> help create ops that submit a Spark job using the Databricks REST API.
+If you want to run a Spark job against YARN or a Spark Standalone cluster, the code inside your asset can issue a shell command invokes `spark-submit`.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:18]:
```suggestion
If you want to run a Spark job against YARN or a Spark Standalone cluster, the code inside your asset can issue a shell command that invokes `spark-submit`.
```

File: docs/content/integrations/spark.mdx [hunk 2]
```diff
@@ -1,62 +1,60 @@
...
 
-If you want to run a Spark job against YARN or a Spark Standalone cluster, you can use <PyObject module='dagster_shell' object='create_shell_command_op' /> to create an op that invokes `spark-submit`.
+This is the easiest approach for migrating existing Spark jobs, and it's the only approach that works for Spark jobs written in Java or Scala. The downside is that it loses out on some of the benefits of Dagster - the implementation of each asset is bound to the execution environment, so you can't run your Spark transformations without relying on external infrastructure. Writing unit tests is cumbersome.
 
-This is the easiest approach for migrating existing Spark jobs, and it's the only approach that works for Spark jobs written in Java or Scala. The downside is that it loses out on some of the benefits of Dagster - the implementation of each op is bound to the execution environment, so you can't run your Spark transformations without relying on external infrastructure. Writing unit tests is cumbersome.
+## Asset accepts and produces DataFrames or RDDs
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:22]: What's an RDD? Can you define it before using it as an acronym?

File: docs/content/integrations/spark.mdx [hunk 3]
```diff
@@ -1,62 +1,60 @@
...
 
-## Op accepts and produces DataFrames or RDDs
+With this approach, the code inside the asset definition consists of pure logical data transformations on Spark DataFrames or RDDs. The `asset`-decorated function accepts `DataFrames` as parameters and returns `DataFrames` when it completes. An [IO manager](/concepts/io-management/io-managers) handles writing and reading the DataFrames to and from persistent storage. The [Running PySpark code in assets](#running-pyspark-code-in-assets) example below shows what this looks like.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:24]: > The `asset`-decorated function

Why use code formatting for asset over leaving it plain or working in the PyObject component for `@asset`?

Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:24]: > An [IO manager](/concepts/io-management/io-managers)

This should be `I/O manager`

File: docs/content/integrations/spark.mdx [hunk 4]
```diff
@@ -1,62 +1,60 @@
...
 
-With this approach, the code inside the op consists of pure logical data transformations on Spark DataFrames or RDDs. The op-decorated function accepts DataFrames as parameters and returns DataFrames when it completes. An [IO manager](/concepts/io-management/io-managers) handles writing and reading the DataFrames to and from persistent storage. The [Running PySpark code in op](#running-pyspark-code-in-ops) example below shows what this looks like.
+If you want your Spark driver to run inside a Spark cluster, you use a "step launcher" resource that informs Dagster how to launch the step. The step launcher resource is responsible for invoking `spark-submit` or submitting the job to Databricks or EMR. [Submitting PySpark jobs on EMR](#submitting-pyspark-jobs-on-emr) shows what this looks like for EMR.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:26]:
```suggestion
If you want your Spark driver to run inside a Spark cluster, you can use a "step launcher" resource that informs Dagster how to launch the step. The step launcher resource is responsible for invoking `spark-submit` or submitting the job to Databricks or EMR. [Submitting PySpark jobs on EMR](#submitting-pyspark-jobs-on-emr) shows what this looks like for EMR.
```

Also, why is step launcher in quotes?

File: docs/content/integrations/spark.mdx [hunk 5]
```diff
@@ -1,62 +1,60 @@
...
 
-If you want your Spark driver to run inside a Spark cluster, you use a "step launcher" resource that informs Dagster how to launch the op. The step launcher resource is responsible for invoking `spark-submit` or submitting the job to Databricks or EMR. [Submitting PySpark ops on EMR](#submitting-pyspark-ops-on-emr) shows what this looks like for EMR.
-
-The advantage of this approach is a very clean local testing story. You can run an entire job of Spark ops in a single process. You can use [IO managers](/concepts/io-management/io-managers) to abstract away IO - storing outputs on the local filesystem during local development and in the cloud in production.
+The advantage of this approach is a very clean local testing story. You can run an entire pipeline of Spark assets in a single process. You can use [IO managers](/concepts/io-management/io-managers) to abstract away IO - storing outputs on the local filesystem during local development and in the cloud in production.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:28]:
```suggestion
The advantage of this approach is a clean local testing story. You can run an entire pipeline of Spark assets in a single process. You can use [I/O managers](/concepts/io-management/io-managers) to abstract away I/O - storing outputs on the local filesystem during local development and in the cloud in production.
```

Suggesting we remove a qualifier (`very`) and format IO as I/O

File: docs/content/integrations/spark.mdx [hunk 6]
```diff
@@ -1,62 +1,60 @@
...
+### Running PySpark code in assets
 
 <CodeReferenceLink filePath="examples/with_pyspark" />
 
-Passing PySpark DataFrames between ops requires a little bit of extra care, compared to other data types, for a couple reasons:
+Passing PySpark DataFrames between asset functions requires a little bit of extra care, compared to other data types, for a couple reasons:
 
 - Spark has a lazy execution model, which means that PySpark won't process any data until an action like `write` or `collect` is called on a DataFrame.
-- PySpark DataFrames cannot be pickled, which means that [IO Managers](/concepts/io-management/io-managers) like the `fs_io_manager` won't work for them.
+- PySpark DataFrames cannot be pickled, which means that [IO Managers](/concepts/io-management/io-managers) like the default filesystem IO manager won't work for them.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:39]:
```suggestion
- PySpark DataFrames cannot be pickled, which means that [I/O managers](/concepts/io-management/io-managers) like the default filesystem I/O manager won't work for them.
```

`IO` > `I/O`

File: docs/content/integrations/spark.mdx [hunk 7]
```diff
@@ -1,62 +1,60 @@
...
 
 In this example, we've defined an <PyObject module='dagster' object='IOManager' /> that knows how to store and retrieve PySpark DataFrames that are produced and consumed by ops.
 
-This example assumes that all the outputs within the dagster job will be PySpark DataFrames and stored in the same way. To learn how to use different IO managers for different outputs within the same dagster job, take a look at the [IO Manager](/concepts/io-management/io-managers) concept page.
+This example assumes that all the outputs within the Dagster pipeline will be PySpark DataFrames and stored in the same way. To learn how to use different IO managers for different outputs within the same Dagster pipeline, take a look at the [IO Manager](/concepts/io-management/io-managers) concept page.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:43]:
```suggestion
This example assumes that all the outputs within the Dagster pipeline will be PySpark DataFrames and stored in the same way. To learn how to use different IO managers for different outputs within the same Dagster pipeline, take a look at the [I/O manager](/concepts/io-management/io-managers) concept page.
```

I know this wasn't part of your changes, but can we please add it in?

File: docs/content/integrations/spark.mdx [hunk 8]
```diff
@@ -1,62 +1,60 @@
...
 
-This example writes out DataFrames to the local file system, but can be tweaked to write to cloud object stores like S3 by changing to the `write` and `read` invocations.
+This example writes out `DataFrames` to the local file system, but can be tweaked to write to cloud object stores like S3 by changing to the `write` and `read` invocations.
```
Comment (erinkcochran87) [inline docs/content/integrations/spark.mdx:45]:
```suggestion
This example writes out `DataFrames` to the local file system, but can be tweaked to write to cloud object stores like Amazon S3 by changing to the `write` and `read` invocations.
```

The first time a branded product is mentioned, the name of the company/creator/provider/etc should be include. Like `Google BigQuery`, thereafter just `BigQuery`.

**Comment by sryza** [comment]
@erinkcochran87 you made a set of comments that I think are all positive improvements and point out real issues with this page, but this PR is currently purely focused on shifting the code examples to focus on assets and making sure that the text still makes sense after that change.  I'd prefer not to expand its scope to focus on general issues with the page.

**Comment by erinkcochran87** [comment]
> general issues with the page

Does this refer to the comments about things like `IO` > `I/O`? If it's that, I'm willing to push those changes off for now. If it's other things I noted about copy that's been updated - ex: explaining acronyms within changed copy - I think the comments are within scope.

**Comment by sryza** [comment]
Sorry, I didn't look closely enough at [this comment](https://github.com/dagster-io/dagster/pull/14537/files#r1213609595) and [this comment](https://github.com/dagster-io/dagster/pull/14537/files#r1213607883), which are directly related to the changes I made. I'll address them.

These non-I/O comments involve copy that's part of lines I touched, but that's independent from the copy I touched:
- https://github.com/dagster-io/dagster/pull/14537#discussion_r1213614550
- https://github.com/dagster-io/dagster/pull/14537/files#r1213608625
- https://github.com/dagster-io/dagster/pull/14537/files#r1213611789

I know I'm being a bit pedantic on this. The background here is that, when I'm looking at the docs, I sometimes see small issues and take a little bit of time out of what I'm "supposed" to be working on to fix them.  This PR is an example of that.  If I know that making those fixes is going to put me on the hook for making additional improvements, it discourages me from making these kinds of incremental improvements in the first place.

File: docs/content/integrations/spark.mdx (hunk 3 above)
Comment (sryza) [reply docs/content/integrations/spark.mdx]: Good point. I fixed this to use PyObject.

File: docs/content/integrations/spark.mdx (hunk 1 above)
Comment (sryza) [reply docs/content/integrations/spark.mdx]: Thanks for catching - fixed this.

**Comment by erinkcochran87** [comment]
Go for it.

**Pull request merged**